src/pymatgen/io/vasp/outputs.py

```diff
@@ -2046,7 +2071,8 @@ def __init__(self, filename: PathLike) -> None:
                     e_wo_entrp = float(match[1])
                 if e0 is None and (match := e0_pattern.search(clean)):
                     e0 = float(match[1])
-            if all([nelect, total_mag is not None, efermi is not None, run_stats]):
+
+            if nelect is not None and total_mag is not None and efermi is not None and run_stats:
```

Comment:
Tag: use truthy to check `run_stats` (dict[str, float | None]) might yield unexpected results as None in dict is also truthy